torchaudio/functional/functional.py

```diff
@@ -177,12 +177,10 @@ def griffinlim(
     # randomly initialize the phase
     batch, freq, frames = specgram.size()
     if rand_init:
-        angles = 2 * math.pi * torch.rand(batch, freq, frames)
+        angles = torch.rand(batch, freq, frames, dtype=torch.cfloat, device=specgram.device)
     else:
-        angles = torch.zeros(batch, freq, frames)
-    angles = torch.stack([angles.cos(), angles.sin()], dim=-1) \
-        .to(dtype=specgram.dtype, device=specgram.device)
-    specgram = specgram.unsqueeze(-1).expand_as(angles)
+        angles = torch.zeros(batch, freq, frames, dtype=torch.cfloat, device=specgram.device)
+        angles.real[:] = 1
```

Comment:
@anjali411 Do you know an elegant and TorchScript-compatible way to initialize Tensors with value `1+0i`?

Reply:
After #52881 is merged, `torch.full(batch, freq, frames, 1j, dtype=torch.cfloat, device=specgram.device)` should work

Reply:
it worked locally. pushing the change.